File: PayPal/src/main/java/com/braintreepayments/api/PayPalAnalytics.java
package com.braintreepayments.api;

enum PayPalAnalytics {

    // Tokenize Events
    TOKENIZATION_STARTED("paypal:tokenize:started"),
    TOKENIZATION_FAILED("paypal:tokenize:failed"),
    TOKENIZATION_SUCCEEDED("paypal:tokenize:succeeded"),
    TOKENIZATION_NETWORK_CONNECTION_FAILED("paypal:tokenize:network-connection:failed"),

    // Browser Presentation Events
    BROWSER_PRESENTATION_SUCCEEDED("paypal:tokenize:browser-presentation:succeeded"),
    BROWSER_PRESENTATION_FAILED("paypal:tokenize:browser-presentation:failed"),

    // Browser Login Events
    BROWSER_LOGIN_CANCELED("paypal:tokenize:browser-login:canceled");

    private final String event;

    PayPalAnalytics(String event) {
        this.event = event;
    }

    String getEvent() {
        return event;
    }
}

File: PayPal/src/main/java/com/braintreepayments/api/PayPalAnalytics.kt
package com.braintreepayments.api

internal object PayPalAnalytics {

    // Conversion Events
    const val TOKENIZATION_STARTED = "paypal:tokenize:started"
    const val TOKENIZATION_FAILED = "paypal:tokenize:failed"
    const val TOKENIZATION_SUCCEEDED = "paypal:tokenize:succeeded"
    const val BROWSER_LOGIN_CANCELED = "paypal:tokenize:browser-login:canceled"

    // Browser Presentation Events
    const val BROWSER_SWITCH_SUCCEEDED = "paypal:tokenize:browser-presentation:succeeded"
    const val BROWSER_SWITCH_FAILED = "paypal:tokenize:browser-presentation:failed"
}
